app/Http/Middleware/Localize.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+
+class Localize
+{
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \Closure $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        $locale = ($request->hasHeader('X-locale')) ? $request->header('X-locale') : app()->config->get('app.locale');
+        app()->setLocale($locale);
```

Comment:
```suggestion
    $request->whenHas('X-Locale', ($locale) => app()->setLocale($locale));
```

or maybe even this would work:

```suggestion
    $request->whenHas('X-Locale', app()->setLocale);
```

Reply:
I didn't know about `whenHas()`. Much nicer. Thank you for suggestion!

Reply:
But at the end non of them works 🤷 😄 
`whenHas` works for `input` (post/get). not for `header`. 
but there is "kind of bug" in the documentation: https://laravel.com/docs/8.x/requests#determining-if-input-is-present
header is clear. but the rest:

> You may use the has method to determine if a value is present on the request.

Is apparently confusing...